### PR TITLE
Adding a base url for relative request

### PR DESCRIPTION
**Pull request by pokap**
Currently, when you want request a link relative of resource, you lost the url entry-point.

I propose to fix it with a base-url on client, and the user can choose the path on entry point.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,9 +29,10 @@ Please provide a version constraint for the ekino/hal-client requirement: dev-ma
 $client = new FileGetContentsHttpClient(array(
     'Authorization' => 'Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ=='
 ));
+$client->setBaseUrl('http://propilex.herokuapp.com');
 
 // create an entry point to retrieve the data
-$entryPoint = new EntryPoint('http://propilex.herokuapp.com', $client);
+$entryPoint = new EntryPoint('/', $client);
```
Comment (rande) [inline README.md:34]: The base url should be the one in the EntryPoint

Comment (pokap) [inline README.md:34]: why? The client make a request on only one API.
Otherwise it will inject the EntryPoint in each Link & Resource.

Comment (rande) [inline README.md:34]: So the base url is mandatory and should be set in the constructor.

File: lib/HttpClient/FileGetContentsHttpClient.php [hunk 2]
```diff
@@ -13,18 +13,47 @@
...
         $this->defaultHeaders = $defaultHeaders;
         $this->timeout = $timeout;
+
+        $this->setBaseUrl($baseUrl);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setBaseUrl($baseUrl)
```
Comment (rande) [inline lib/HttpClient/FileGetContentsHttpClient.php]: don't expose this settings. This should never change.

File: lib/HttpClient/FileGetContentsHttpClient.php [hunk 3]
```diff
@@ -52,10 +75,20 @@ protected function doRequest($url, $method, array $headers, array $data)
             )
         );
 
+        // if is relative url
+        if ('http' !== substr($url, 0, 4)) {
```
Comment (rande) [inline lib/HttpClient/FileGetContentsHttpClient.php:79]: I missed this point, you need to handle https ;)

Comment (rande) [inline lib/HttpClient/FileGetContentsHttpClient.php:79]: forget ;)

**Pull request merged**